orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/ManifestLabelValueExpressionFunctionProvider.java

```diff
@@ -87,7 +87,11 @@ public static String manifestLabelValue(
           "A valid Deploy Manifest stage name is required for this function");
     }
 
-    List<Map> manifests = (List<Map>) stage.get().getContext().get("manifests");
+    // using outputs.manifests to give access to labels added by Spinnaker itself during the
+    // deployment
+    // this is safe as we assert above that we're only using successful deploy stages, so this key
+    // should always exist
+    List<Map> manifests = (List<Map>) stage.get().getContext().get("outputs.manifests");
```

Comment:
Is this safe timing-wise?  Like, could this run when manifests exists, but output.manifests doesn't?

Reply:
I'm reasonably confident `outputs.manifests` should always exist once a deploy stage is succeeded - and this expression only considers stages in a succeeded state.

Reply:
Aaah, I see it [a few lines up](https://github.com/spinnaker/orca/pull/4508/files#diff-275c694ae6c05141ecb04851930851ddefd1d55195336b0f3c5ac1e46639f758R82).